core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -1,31 +1,39 @@
 package com.databricks.labs.remorph.parsers.tsql
 
-import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
-import com.databricks.labs.remorph.parsers.intermediate.{Column, Expression, Literal}
+import org.antlr.v4.runtime.Token
 import org.antlr.v4.runtime.tree.TerminalNode
+
+import com.databricks.labs.remorph.parsers.intermediate.{Column, Literal}
 import com.databricks.labs.remorph.parsers.tsql.TSqlParser._
+import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
 
 class TSqlExpressionBuilder
-    extends TSqlParserBaseVisitor[Expression]
+    extends TSqlParserBaseVisitor[ir.Expression]
     with ParserCommon
     with IncompleteParser[ir.Expression] {
 
   protected override def wrapUnresolvedInput(unparsedInput: String): ir.UnresolvedExpression =
     ir.UnresolvedExpression(unparsedInput)
 
-  override def visitSelect_list_elem(ctx: Select_list_elemContext): Expression = ctx.expression_elem().accept(this)
+  override def visitFull_column_name(ctx: Full_column_nameContext): ir.Expression = {
+    Column(ctx.id_.getText)
+  }
 
-  override def visitExpression(ctx: ExpressionContext): Expression = ctx match {
-    case c if c.full_column_name() != null => c.full_column_name().accept(this)
-    case c if c.primitive_expression() != null => c.primitive_expression().accept(this)
+  // This is merely a placeholder until the grammar can be fixed. It ensures that the
+  // precedence expression is correctly handled
+  override def visitBracket_expression(ctx: Bracket_expressionContext): ir.Expression = {
+    ctx.expression().accept(this)
```

Comment:
What about `ctx.subquery`? Will it be handled in a future change ?

Reply:
Yes - IN fact this rule is wrong and I need to fix the grammar in another PR